reactioncommerce/reaction-admin
# Product editor fixes

Resolves #237
Impact: **minor**
Type: **feature|bugfix**

## Issue

There are various issues with the product editor outlined in this ticket #237 

## Solution

**Fixed**
- Variants in the sidebar are highlighted when active
- Inconsistent notification. Now all toasts use notistack
- Archived variants and options seem to work. Not sure if I fixed it, or if some other issue was at play
- Refactor GQL queries/mutations to use fragments and moved out of hook file
- In the `useProduct` hook, skip queries that require `shopId` if the `shopId` is not available

**Not fixed**
- Form saving issue has been pushed off to this ticket: https://github.com/reactioncommerce/reaction-admin/issues/244

## Breaking changes

none

## Testing
1. Ensure variants and options are highlighted
1. Check the code for the queries/mutations
1. Ensure variants and options can be archived
